src/main/scala/day5/rock_paper_scissors/Main.scala

```diff
@@ -0,0 +1,164 @@
+package day5.rock_paper_scissors
+import scala.collection.mutable.ArrayBuffer
+import scala.util.Random
+
+
+object Main extends App {
+  val weights = Array(2,2,2,2,2)
+  var wins = 0
+  var losses = 0
+  val previous = ArrayBuffer.empty[String]
+
+  def userChoice():String ={
+    scala.io.StdIn.readLine("Rock, Paper, Scissors, Lizard, Spock? ").toString
+  }
+
+  def validate(choice:String):Boolean = {
+    choice match {
+      case "Rock" | "rock" => true
+      case "Paper" | "paper" => true
+      case "Scissors" | "scissors" => true
+      case "Lizard" | "lizard" => true
+      case "Spock" | "spock" => true
+      case _ => false
+    }
+  }
+
+  def aiChoice(): String = {
+    val choices = Array("Rock", "Paper", "Scissors", "Lizard", "Spock")
+    choices(Random.nextInt(choices.length))
+  }
+
+  def weightedAI(weights: Array[Int]): String = {
+    var num = Random.nextInt(weights.sum)
+    val choices = Array("Rock", "Paper", "Scissors", "Lizard", "Spock")
+    for (i <- 0 until 5) {
+      num -= weights(i)
+      if (num <= 0) {
+        return choices(i)
+      }
+    }
+    "Spock"
+  }
+
+  def patternAI(previous: ArrayBuffer[String]):String = {
+    val choices = Array("Rock", "Paper", "Scissors", "Lizard", "Spock")
+    if (previous.length>=5) {
+
+    }
+    else {
+      choices(Random.nextInt(choices.length))
+    }
+
+    ""
+  }
+
```

Comment:
Consider rewriting the methods for rock, paper etc. into one method that takes in something another parameter, something like:

def foo(playerChoice: String, other: String, weight: Array[Int])

have another tester function that checks if playerChoice wins against other
then for the fancy print-lines, you can add some generic things elsewhere, e.g.

winPrinter(winner: String, loser: String): String = match winner {
  case "Rock" => s"$winner crushes ${loser}!
  ...
}